File: python/models/diplomado.py
from .programa_academico import ProgramaAcademico

class Diplomado(ProgramaAcademico):

    def __init__(self):
        super().__init__("Diplomado")

    def evaluar_aprobacion(self, notas):
        promedio = sum(notas) / len(notas)
        return promedio >= 14

File: java/Diplomado.java
import java.util.List;

public class Diplomado extends ProgramaAcademico {
    private final double notaMinima = 14.0;

    public Diplomado() {
        super("Diplomado");
    }

    @Override
    public boolean evaluarAprobacion(List<Double> notas) {
        return !notas.isEmpty() && notas.stream().mapToDouble(Double::doubleValue).average().orElse(0) >= notaMinima;
    }

    @Override
    public String obtenerCriterio() {
        return "promedio mayor o igual a 14/20";
    }
}
